x-pack/plugins/security_solution/public/common/hooks/use_invalid_filter_query.test.tsx

```diff
@@ -0,0 +1,286 @@
+      {children}
+    </TestProviders>
+  );
+};
+
+describe('useInvalidFilterQuery', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('invokes error toast with error title and error instance without original stack', () => {
+    const store = getStore();
+    const props = getProps();
+
+    renderHook(useInvalidFilterQuery, { initialProps: props, wrapper: getWrapper(store) });
+
+    expect(store.getState().app.errors).toEqual([
+      {
+        displayError: true,
+        id: props.id,
+        hash: genHash(props.kqlError.message),
+        message: [props.kqlError.message],
+        title: props.kqlError.name,
+      },
+    ]);
+
+    expect(kibanaMock.notifications.toasts.addError).toHaveBeenCalledTimes(1);
+    expect(kibanaMock.notifications.toasts.addError).toHaveBeenCalledWith(props.kqlError, {
+      title: props.kqlError.name,
+    });
+    expect(props.kqlError.stack).toBeUndefined();
+  });
+
+  it('does not invoke error toast, when kqlError is missing', () => {
+    const store = getStore();
+    const props = getProps();
+
+    renderHook(useInvalidFilterQuery, {
+      initialProps: {
+        id: props.id,
+        query: props.query,
+        startDate: props.startDate,
+        endDate: props.endDate,
+      },
+      wrapper: getWrapper(store),
+    });
+
+    expect(kibanaMock.notifications.toasts.addError).not.toHaveBeenCalled();
+  });
+
+  it('does not invoke error toast, when kqlError misses name property', () => {
+    const store = getStore();
+    const props = getProps();
+
+    // @ts-expect-error
+    props.kqlError.name = null;
+    renderHook(useInvalidFilterQuery, { initialProps: props, wrapper: getWrapper(store) });
+
+    expect(kibanaMock.notifications.toasts.addError).not.toHaveBeenCalled();
+  });
+
+  it('does not invoke error toast, when kqlError misses message property', () => {
+    const store = getStore();
+    const props = getProps();
+
+    // @ts-expect-error
+    props.kqlError.message = null;
+    renderHook(useInvalidFilterQuery, { initialProps: props, wrapper: getWrapper(store) });
+
+    expect(kibanaMock.notifications.toasts.addError).not.toHaveBeenCalled();
+  });
+
+  it('does not invoke error toast, when filterQuery is present', () => {
+    const store = getStore();
+    const props = getProps();
+
+    renderHook(useInvalidFilterQuery, {
+      initialProps: { ...props, filterQuery: 'filterQuery' },
+      wrapper: getWrapper(store),
+    });
+
+    expect(kibanaMock.notifications.toasts.addError).not.toHaveBeenCalled();
+  });
+
+  // POTENTIAL BUG:
+  //
+  // id should ensure that only one toast is shown
+  // when you have different errors with the same id,
+  // but this test shows that it will currently show multiple toasts
+  // so we need to double check if this is the intended behavior and fix otherwise
+  it('invokes toast for each error, when called multiple times with same id and different errors, during a single render', async () => {
```

Comment:
I have not been able to reproduce it.. For example, in the demo below, I see multiple errors to the query but I see only one toast. Do you think it might be worth checking why that is the case.

Reply:
Hey Jatin, what do you mean by "demo below"?

Reply:
@kapral18 , Sorry I missed attaching video in my last comment. Just added it.